crates/ruff_python_parser/src/parser/pattern.rs

```diff
@@ -129,13 +133,10 @@ impl<'src> Parser<'src> {
             };
 
             let lhs_value = if let Pattern::MatchValue(lhs) = lhs {
-                if !lhs.value.is_literal_expr() && !matches!(lhs.value.as_ref(), Expr::UnaryOp(_)) {
+                if !matches!(&*lhs.value, Expr::NumberLiteral(_) | Expr::UnaryOp(_)) {
```

Comment:
If the current token is either a `+` or `-`, then the LHS can only be either a number or unary op with a number as stated in the grammar.